packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -386,12 +386,10 @@ export const DatePickerMixin = (subclass) =>
           }
         }
 
-        if (this.inputElement.value === '' && this.__dispatchChange) {
```

Comment:
Changed the logic to not use `__dispatchChange` flag here, as we actually only need to check if the `<input>` value has been cleared by the user, and update the date-picker `value` property accordingly.